elmslie-core/src/main/java/vivid/money/elmslie/core/config/ElmslieConfig.kt

```diff
@@ -2,30 +2,42 @@ package vivid.money.elmslie.core.config
 
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.asCoroutineDispatcher
 import vivid.money.elmslie.core.logger.ElmslieLogConfiguration
 import vivid.money.elmslie.core.logger.ElmslieLogger
 import vivid.money.elmslie.core.logger.strategy.IgnoreLog
+import java.util.concurrent.Executors
 
 object ElmslieConfig {
 
-    @Volatile private lateinit var _logger: ElmslieLogger
+    @Volatile
+    private lateinit var _logger: ElmslieLogger
 
-    @Volatile private lateinit var _ioDispatchers: CoroutineDispatcher
+    @Volatile
+    private lateinit var _ioDispatchers: CoroutineDispatcher
 
-    @Volatile private var _shouldStopOnProcessDeath: Boolean = true
+    @Volatile
+    private lateinit var _eventDispatchersFactory: () -> CoroutineDispatcher
+
+    @Volatile
+    private var _shouldStopOnProcessDeath: Boolean = true
 
     val logger: ElmslieLogger
         get() = _logger
 
     val ioDispatchers: CoroutineDispatcher
         get() = _ioDispatchers
 
+    val eventDispatchers: CoroutineDispatcher
+        get() = _eventDispatchersFactory.invoke()
+
     val shouldStopOnProcessDeath: Boolean
         get() = _shouldStopOnProcessDeath
 
     init {
         logger { always(IgnoreLog) }
         ioDispatchers { Dispatchers.IO }
+        eventDispatchers { { Executors.newFixedThreadPool(1).asCoroutineDispatcher() } }
```

Comment:
I think we shouldn't let users set the custom dispatchers over here since it's an internal requirement of the library to handle event's stream one by one as a queue.
Letting set the custom one, user can easily break the behavior...

Reply:
But from another side maybe there are more convenient ways to create a single thread dispatcher inside the application. For example custom thread pool, theoretically, it can help to avoid the creation of threads for each store.

Reply:
Premature optimization =)